app/controllers/schools/courses_controller.rb

```diff
@@ -176,16 +176,32 @@ def assignments
       @milestones =
         @course
           .targets
+          .includes(:assignments)
           .live
           .milestone
           .order("assignments.milestone_number asc")
           .page(params[:page])
           .per(20)
       @page_no = params[:page].presence || 1
+      @have_gaps = gap?(@milestones.pluck("assignments.milestone_number"))
     end
 
     private
 
+    def gap?(arr)
+      arr.sort! # Sort the array in ascending order
+      n = arr.length
+
+      # Check if there is a gap between consecutive elements
+      (0..n - 2).each do |i|
+        if arr[i + 1] - arr[i] > 1
+          return true # Gap found
+        end
+      end
+
+      return false # No gaps found
+    end
```

Comment:
- `sort!` mutates the array it's called on - it's usually not a good choice.
- This method can be much simpler.

```suggestion
    def gap?(numbers)
      remains = (1..numbers.length).to_a - numbers
      remains.present?
    end
```